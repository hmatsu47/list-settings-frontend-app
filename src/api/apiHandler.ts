import { fetchWithTimeout } from "./fetchWithTimeout";

// ベース URI
export const baseUri = "/api";
export const baseUriTag = "/api";
export const remoteBaseUri = localStorage.getItem("remoteBaseUri");
export const remoteBaseUriTag = localStorage.getItem("remoteBaseUriTag");

// API からデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout("GET", url)).json();
};
