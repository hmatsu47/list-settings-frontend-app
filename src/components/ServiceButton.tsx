import Button from "@suid/material/Button";
import { service, setService } from "../signal";

type Props = {
  serviceName: string;
};

export const ServiceButton = (props: Props) => {
  return (
    <Button
      variant={service() === props.serviceName ? "contained" : "outlined"}
      size="small"
      color={service() === props.serviceName ? "primary" : "inherit"}
      onClick={(e) => {
        setService(props.serviceName);
        // 選択肢をローカルストレージに記録しておく
        localStorage.setItem("selectedService", props.serviceName);
      }}
      sx={{ textTransform: "none" }}
      title={`${props.serviceName} を選択`}
    >
      {props.serviceName}
    </Button>
  );
};
