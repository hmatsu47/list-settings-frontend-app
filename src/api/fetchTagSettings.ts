import { ErrorResponse, TagSetting } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import {
  message,
  messageSeverity,
  setMessage,
  setMessageSeverity,
  setTagSettings,
} from "../signal";

export const fetchTagSettings = async () => {
  const load = async (): Promise<void> => {
    const data: TagSetting[] | ErrorResponse = await getApiData(
      `${baseUri}/tagSettings`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setTagSettings(undefined);
      // すでにエラーメッセージがセットされている場合は連結する（複数 API を続けて呼び出すため）
      const tempMessage =
        message() && messageSeverity() === "error" ? message() : "";

      setMessage(`${tempMessage}${(data as ErrorResponse).message}`);
      setMessageSeverity("error");
      return;
    }
    setTagSettings(data as TagSetting[]);
    // エラーメッセージは上書きしない
  };
  void load();
};
