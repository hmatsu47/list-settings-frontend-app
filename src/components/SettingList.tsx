import { onMount } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import {
  fetchLocal,
  fetchRemote,
  fetchUriSettings,
} from "../api/fetchUriSettings";
import { fetchTagSettings } from "../api/fetchTagSettings";
import { tagSettings, uriSettings, uriRemoteSettings } from "../signal";
import { TagSettingList } from "./TagSettingList";
import { UriSettingList } from "./UriSettingList";

export const SettingList = () => {
  onMount(() => {
    if (uriSettings() === undefined) {
      fetchUriSettings(fetchLocal);
    }
    if (uriRemoteSettings() === undefined) {
      fetchUriSettings(fetchRemote);
    }
    if (tagSettings() === undefined) {
      fetchTagSettings();
    }
  });

  return (
    <Stack spacing={2} direction="column">
      <UriSettingList />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      />
      <TagSettingList />
    </Stack>
  );
};
