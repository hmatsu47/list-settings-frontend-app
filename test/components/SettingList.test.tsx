// @vitest-environment jsdom

import { describe, expect, test, vi } from "vitest";
import "vi-fetch/setup";
import { mockFetch, mockGet } from "vi-fetch";
import { render } from "@solidjs/testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { SettingList } from "../../src/components/SettingList";
import { baseUri, remoteBaseUri } from "../../src/api/apiHandler";
import {
  setService,
  setServices,
  setTagSettings,
  setUriSettings,
  setUriRemoteSettings,
} from "../../src/signal";
import {
  LastReleased,
  NextRelease,
  TagSetting,
  UriSetting,
} from "../../src/type";

describe("<SettingList />", () => {
  const settingList = [
    {
      title: "未fetch",
      fetched: false,
      uriSettings: [
        {
          environment_name: "test",
          last_released: {
            image_uri: "testtesttest",
            released_at: new Date("2023-05-14T22:05:00+0900"),
          } as LastReleased,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
        {
          environment_name: "stg",
          last_released: {
            image_uri: "foobar",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: {
            image_uri: "foobarbaz",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      remoteSettings: [
        {
          environment_name: "prod",
          last_released: null,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-17T23:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      tagSettings: [
        { environment_name: "dev", tags: ["hogefuga", "latest"] } as TagSetting,
        { environment_name: "prod", tags: ["release", "foobar"] } as TagSetting,
      ],
      selectedService: undefined,
      expectedButtonLabel: null,
      expected: [
        { label: "リリース対象サービスが未選択です", count: 1 },
        { label: "リポジトリにイメージがありません", count: 1 },
      ],
      expectedTitle: null,
    },
    {
      title:
        "サービス選択なし・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリにレコードなし",
      fetched: true,
      uriSettings: null,
      remoteSettings: null,
      tagSettings: null,
      selectedService: undefined,
      expectedButtonLabel: null,
      expected: [
        { label: "リリース対象サービスが未選択です", count: 1 },
        { label: "リポジトリにイメージがありません", count: 1 },
      ],
      expectedTitle: null,
    },
    {
      title:
        "サービス選択なし・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリに1レコードあり",
      fetched: true,
      uriSettings: null,
      remoteSettings: null,
      tagSettings: [
        {
          environment_name: "dev",
          tags: ["tag-hogefuga", "latest"],
        } as TagSetting,
      ],
      selectedService: undefined,
      expectedButtonLabel: [{ label: "開発環境設定", count: 1 }],
      expected: [{ label: "リリース対象サービスが未選択です", count: 1 }],
      expectedTitle: null,
    },
    {
      title:
        "サービス選択あり・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリにレコードなし",
      fetched: true,
      uriSettings: null,
      remoteSettings: null,
      tagSettings: null,
      selectedService: "test1",
      expectedButtonLabel: null,
      expected: [{ label: "リポジトリにイメージがありません", count: 2 }],
      expectedTitle: null,
    },
    {
      title:
        "サービス選択あり・URI形式両側のリポジトリにレコードあり・タグ形式リポジトリに1レコードあり",
      fetched: true,
      uriSettings: [
        {
          environment_name: "test",
          last_released: {
            image_uri: "testtesttest",
            released_at: new Date("2023-05-14T22:05:00+0900"),
          } as LastReleased,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
        {
          environment_name: "stg",
          last_released: {
            image_uri: "foobar",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: {
            image_uri: "foobarbaz",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      remoteSettings: [
        {
          environment_name: "prod",
          last_released: null,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-17T23:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      tagSettings: [
        { environment_name: "dev", tags: ["hogefuga", "latest"] } as TagSetting,
        { environment_name: "prod", tags: ["release", "foobar"] } as TagSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: [
        { label: "test環境設定", count: 1 },
        { label: "ステージング環境設定", count: 1 },
        { label: "本番環境設定", count: 2 },
        { label: "開発環境設定", count: 1 },
      ],
      expected: [
        { label: "前回：testtesttest", count: 1 },
        { label: "前回：2023-05-14 22:05", count: 1 },
        { label: "次回：foobarbaz", count: 1 },
        { label: "次回：2023-05-15 04:05", count: 1 },
        { label: "次回：hogefuga", count: 1 },
        { label: "次回：2023-05-17 23:05", count: 1 },
        { label: "hogefuga, latest", count: 1 },
        { label: "release, foobar", count: 1 },
      ],
      expectedTitle: ["前回：foobar | 2023-05-11 04:05"],
    },
  ];
  beforeEach(() => {
    // localStorage はモック化すべきかも（うまくいかなかったため一旦モックなしで実装）
    localStorage.setItem("baseUri", "/api");
    localStorage.setItem("remoteBaseUri", "http://remote.example.com/api");
    localStorage.removeItem("selectedService");
    localStorage.setItem(
      "uriButtonColor",
      '{"test":"#2c387e","stg":"#00695f","prod":"#e91e63"}'
    );
    localStorage.setItem("uriSettingsHeaderTitle", "URI形式ヘッダータイトル");
    localStorage.setItem("uriSettingUriPrefix", "http://");
    localStorage.setItem("uriSettingUriSuffix", "-test.example.com");
    localStorage.setItem("tagSettingsHeaderTitle", "タグ形式ヘッダータイトル");
    localStorage.setItem("tagSettingUriPrefix", "http://");
    localStorage.setItem("tagSettingUriSuffix", "-test.example.com");
    mockFetch.clearAll();
  });
  afterEach(() => {
    vi.clearAllMocks();
    localStorage.removeItem("baseUri");
    localStorage.removeItem("remoteBaseUri");
    localStorage.removeItem("selectedService");
    localStorage.removeItem("uriButtonColor");
    localStorage.removeItem("uriSettingsHeaderTitle");
    localStorage.removeItem("uriSettingUriPrefix");
    localStorage.removeItem("uriSettingUriSuffix");
    localStorage.removeItem("tagButtonColor");
    localStorage.removeItem("tagSettingsHeaderTitle");
    localStorage.removeItem("tagSettingUriPrefix");
    localStorage.removeItem("tagSettingUriSuffix");
  });
  settingList.forEach((testCase) => {
    test(testCase.title, () => {
      const mockTag = mockGet(`${baseUri}/tagSettings`).willResolve(
        testCase.tagSettings
      );
      const mockUri = mockGet(`${baseUri}/uriSettings`).willResolve(
        testCase.uriSettings
      );
      const mockUriRemote = mockGet(`${remoteBaseUri}/uriSettings`).willResolve(
        testCase.remoteSettings
      );
      setServices(["test1", "test2"]);
      if (testCase.selectedService) {
        localStorage.setItem("selectedService", testCase.selectedService);
      }
      setService(testCase.selectedService);
      if (testCase.fetched) {
        setUriSettings(testCase.uriSettings);
        setUriRemoteSettings(testCase.remoteSettings);
        setTagSettings(testCase.tagSettings);
      } else {
        setUriSettings(undefined);
        setUriRemoteSettings(undefined);
        setTagSettings(undefined);
      }
      const { container, getByText, getAllByText, getByTitle, unmount } =
        render(() => <SettingList />);
      // css の名前が動的に変わるので固定値に置換
      const html = formatSnapshot(container.innerHTML);
      expect(html).toMatchSnapshot();
      // fetch 後かどうかで分岐
      if (testCase.fetched) {
        // 各ボタン（クリックしない）
        if (testCase.expectedButtonLabel) {
          testCase.expectedButtonLabel.forEach((buttonLabel) => {
            if (buttonLabel.count === 1) {
              const text = getByText(buttonLabel.label) as HTMLElement;
              expect(text).toHaveTextContent(buttonLabel.label);
            } else {
              const text: string[] | any[] = getAllByText(
                buttonLabel.label
              ) as HTMLElement[];
              expect(text.length).toBe(buttonLabel.count);
            }
          });
        }
        // 各 URI・日付・タグ
        testCase.expected.forEach((expected) => {
          if (expected.count === 1) {
            const text = getByText(expected.label) as HTMLElement;
            expect(text).toHaveTextContent(expected.label);
          } else {
            const text: string[] | any[] = getAllByText(
              expected.label
            ) as HTMLElement[];
            expect(text.length).toBe(expected.count);
          }
        });
        // title
        if (testCase.expectedTitle) {
          testCase.expectedTitle.forEach((title) => {
            const text = getByTitle(title) as HTMLElement;
            expect(text).toBeInTheDocument();
          });
        }
      } else {
        // とりあえず API の呼び出しが行われたことだけを確認（戻り値は今のところ上手くテストできず）
        expect(mockTag).toHaveFetched();
        expect(mockUri).toHaveFetched();
        expect(mockUriRemote).toHaveFetched();
      }
      unmount();
    });
  });
});
