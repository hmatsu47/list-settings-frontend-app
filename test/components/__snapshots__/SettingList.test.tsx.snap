// Vitest Snapshot v1

exports[`<SettingList /> > サービス選択あり・URI形式両側のリポジトリにレコードあり・タグ形式リポジトリに1レコードあり 1`] = `"<div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">URI形式ヘッダータイトル</h5><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービス</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test1 を選択\\" type=\\"button\\">test1<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test2 を選択\\" type=\\"button\\">test2<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">設定　　　　　　　　　　</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">イメージURI　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">リリース日時　　　　　　</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test環境のリリース設定画面を開く\\" type=\\"button\\">test環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：testtesttest</p></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：2023-05-14 22:05</p></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"ステージング環境のリリース設定画面を開く\\" type=\\"button\\">ステージング環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：foobarbaz<span class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiIconButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" aria-label=\\"info\\" title=\\"前回：foobar | 2023-05-11 04:05\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root css-xxxxxxxx\\" data-testid=\\"InfoIcon\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\"><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z\\"></path></svg><span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></span></p></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：2023-05-15 04:05</p></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：hogefuga</p></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：2023-05-17 23:05</p></td></tr></tbody></table></div></div></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">タグ形式ヘッダータイトル</h5><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">設定　　　　　　　　　　</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">リリース対象イメージタグ　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"開発環境のリリース設定画面を開く\\" type=\\"button\\">開発環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">hogefuga, latest</p></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">release, foobar</p></td></tr></tbody></table></div></div></div></div>"`;

exports[`<SettingList /> > サービス選択あり・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリにレコードなし 1`] = `"<div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">URI形式ヘッダータイトル</h5><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービス</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test1 を選択\\" type=\\"button\\">test1<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test2 を選択\\" type=\\"button\\">test2<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">タグ形式ヘッダータイトル</h5><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div></div></div>"`;

exports[`<SettingList /> > サービス選択なし・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリに1レコードあり 1`] = `"<div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">URI形式ヘッダータイトル</h5><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービス</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test1 を選択\\" type=\\"button\\">test1<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test2 を選択\\" type=\\"button\\">test2<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービスが未選択です</h6></div></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">タグ形式ヘッダータイトル</h5><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">設定　　　　　　　　　　</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">リリース対象イメージタグ　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　　</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"開発環境のリリース設定画面を開く\\" type=\\"button\\">開発環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">tag-hogefuga, latest</p></td></tr></tbody></table></div></div></div></div>"`;

exports[`<SettingList /> > サービス選択なし・URI形式両側のリポジトリにレコードなし・タグ形式リポジトリにレコードなし 1`] = `"<div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">URI形式ヘッダータイトル</h5><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービス</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test1 を選択\\" type=\\"button\\">test1<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"test2 を選択\\" type=\\"button\\">test2<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リリース対象サービスが未選択です</h6></div></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"></div><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-root css-xxxxxxxx\\">タグ形式ヘッダータイトル</h5><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div></div></div>"`;

exports[`<SettingList /> > 未fetch 1`] = `"<div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"></div></div>"`;
